app/src/main/java/pl/pwr/zpi/bcycle/mobile/TripCreationMapActivity.kt

```diff
@@ -0,0 +1,326 @@
+        setContentView(R.layout.activity_trip_creation_map)
+        mapFragment = supportFragmentManager
+            .findFragmentById(R.id.myMap) as SupportMapFragment
+        mapFragment.getMapAsync(this)
+        locationManager = getSystemService(Context.LOCATION_SERVICE) as LocationManager
+        setListeners()
+        getSavedData()
+    }
+
+    private fun getSavedData() {
+        val extras = intent.extras;
+        savedDesc = extras.getString(TripCreationActivity.DESCRIPTION_KEY)!!
+        savedName = extras.getString(TripCreationActivity.NAME_KEY)!!
+        savedEndDate = extras.getSerializable(TripCreationActivity.END_DATE_KEY)!! as ZonedDateTime
+        savedStartDate =
+            extras.getSerializable(TripCreationActivity.START_DATE_KEY)!! as ZonedDateTime
+    }
+
+    private fun setListeners() {
+        bt_next.setOnClickListener {
+            LovelyStandardDialog(this)
+                .setTopColorRes(R.color.violet)
+                .setTitle(resources.getString(R.string.prompt_is_trip_done))
+                .setIcon(R.drawable.bike_icon)
+                .setPositiveButton(R.string.yes) {
+                    if(isRequiredDataGiven()){
+                        val photoUrl = FirebaseStorage.getInstance()
+                            .reference.child(auth.currentUser!!.uid + ".png").toString()
+                        val route = createMarkersList()
+                        ApiClient.groupTripApi.create(
+                            GroupTrip(
+                                null, savedName, savedDesc,
+                                UserInfo(
+                                    auth.currentUser?.uid!!,
+                                    auth.currentUser?.displayName!!,
+                                    auth.currentUser?.email!!,
+                                    photoUrl
+                                ), null, savedStartDate, savedEndDate, route, null
+                            )
+                        ).background().subscribe({
+                            showToast(getString(R.string.prompt_trid_added_successfully))
+                            startActivity(
+                                Intent(
+                                    this,
+                                    MainActivity::class.java
+                                ).addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+                            )
+                            finish()
+                        }, { showToast(it.message?:"mess null")})
+                    }
+                    else{
+                        showToastError(R.string.error_missing_start_or_end_marker)
+                    }
+                }
+                .setPositiveButtonColorRes(R.color.green)
+                .setNegativeButton(R.string.no, {})
+                .setNegativeButtonColorRes(R.color.red)
+                .show()
+        }
+        bt_show_start.setOnClickListener {
+            if (markerStartPoint != null) {
```

Comment:
Move handlers to separate methods. `bt_show_start` and `bt_show_end` can use the same handler method with an argument.